packages/sdk/src/libs/clients/lsp/client.e2e.ts

```diff
@@ -0,0 +1,22 @@
+require("dotenv").config();
+import assert from "assert";
+import * as Client from "./client";
+import { ethers } from "ethers";
+
+const address = "0x651EcbFc3d03109Bb9B2183A068F61dF6935a15A";
+// these require integration testing, skip for ci
+describe("lsp", function () {
```

Comment:
do you plan on adding more tests? for example, validating each possible event is captured and stored correctly?

Reply:
yea i could, im doing this more as an integration test, ie it reaches out to a real contract and builds up the state.  I could though do this manually by crafting the events. I can give it a try, it might be a better approach and more unit test like.

Reply:
ok i spent almost a whole day on trying to stub out events so i could properly unit test. it turns out its not possible to create the event type without a provider. so the only way to really do this is to deploy everything to a local node. its pretty involved, so probably want an independent task for this!

Reply:
https://app.clubhouse.io/uma-project/story/1654/add-unit-tests-against-local-blockchain-when-needed

Reply:
that sounds good. sorry for waisting time on that

Reply:
all good, im glad i tried it, learned quite a bit